v0/destinations/hs/transform.js

```diff
@@ -91,30 +91,46 @@ const processRouterDest = async inputs => {
     const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
     return [respEvents];
   }
+
   const successRespList = [];
   const errorRespList = [];
   // using the first destination config for transforming the batch
   const { destination } = inputs[0];
   let propertyMap;
   const mappedToDestination = get(inputs[0].message, MappedToDestinationKey);
-  const { objectType } = getDestinationExternalIDInfoForRetl(inputs[0].message, "HS");
-  if (
-    mappedToDestination &&
-    GENERIC_TRUE_VALUES.includes(mappedToDestination?.toString())
-  ) {
-    // skip splitting the batches to inserts and updates if object it is an association
-    if (objectType.toLowerCase() !== "association") {
-      // get info about existing objects and splitting accordingly.
-      inputs = await splitEventsForCreateUpdate(inputs, destination);
-    }
-  } else {
-    // reduce the no. of calls for properties endpoint
-    const traitsFound = inputs.some(input => {
-      return fetchFinalSetOfTraits(input.message) !== undefined;
-    });
-    if (traitsFound) {
-      propertyMap = await getProperties(destination);
+  const { objectType } = getDestinationExternalIDInfoForRetl(
+    inputs[0].message,
+    "HS"
+  );
+
+  try {
+    if (
+      mappedToDestination &&
+      GENERIC_TRUE_VALUES.includes(mappedToDestination?.toString())
+    ) {
+      // skip splitting the batches to inserts and updates if object it is an association
+      if (objectType.toLowerCase() !== "association") {
+        // get info about existing objects and splitting accordingly.
+        inputs = await splitEventsForCreateUpdate(inputs, destination);
+      }
+    } else {
+      // reduce the no. of calls for properties endpoint
+      const traitsFound = inputs.some(input => {
+        return fetchFinalSetOfTraits(input.message) !== undefined;
+      });
+      if (traitsFound) {
+        propertyMap = await getProperties(destination);
+      }
     }
+  } catch (error) {
+    errorRespList.push(
+      getErrorRespEvents(
+        [inputs[0].metadata],
+        error.response ? error.response.status : 400,
+        error.message || "Error occurred while processing payload."
+      )
+    );
+    return [...errorRespList];
```

Comment:
Can we rather return `errorRespList` instead of destructuring the error responses ?